web/src/lib/utils/licenseStatus.test.ts

```diff
@@ -32,4 +40,23 @@ describe("licenseStatus", () => {
       });
     });
   });
+
+  describe("licenseDataModifyingFunction", () => {
+    it("returns a fn that updates the business that was passed in as the argument when invoked", () => {
+      const business = generateBusiness({ id: "id 1", licenseData: undefined });
+
+      const licenses = {
+        [randomElementFromArray(Object.values(taskIdLicenseNameMapping))]: {
+          ...generateLicenseDetails({}),
+        },
+      };
+
+      const userDataInDb = generateUserDataForBusiness(
+        generateBusiness({ id: "id 1", licenseData: generateLicenseData({ licenses }) })
+      );
+      const fn = licenseDataModifyingFunction(userDataInDb, "id 1");
+      const result = (fn(business).licenseData as LicenseData).licenses;
```

Comment:
I'm not quite sure how to resolve this if we are not using the as keyword, when I remove it, typescript is screaming at me saying that license date I can be undefined, but the test are passing